### PR TITLE
refactor: explicit deny alert when version not match

### PR DESCRIPTION
**Pull request by jjyr**

### CODE REVIEW

File: util/network-alert/src/notifier.rs [hunk 1]
```diff
@@ -44,23 +44,26 @@ impl Notifier {
...
-        {
-            return;
+            if let Some(ref max_v) = alert.max_version {
+                if Version::parse(&max_v)
+                    .map(|max_v| client_version > max_v)
+                    .unwrap_or(true)
+                {
+                    return;
+                }
+            }
```
Comment (keroro520) [inline util/network-alert/src/notifier.rs]: It's better to define a helper function, like `is_allowed_version` or else.

Comment (keroro520) [reply util/network-alert/src/notifier.rs]: Please add unit tests.

Comment (jjyr) [reply util/network-alert/src/notifier.rs]: the old test case is works well https://github.com/nervosnetwork/ckb/blob/develop/util/network-alert/src/tests/test_notifier.rs#L6

this PR just explicit the program's behaviour

**Comment by u2** [comment]
bors ping

File: util/network-alert/src/notifier.rs [hunk 2]
```diff
@@ -30,9 +30,32 @@ impl Notifier {
         }
     }
 
-    pub fn add(&mut self, alert: Arc<Alert>) {
+    fn is_accept_version(&self, alert: &Arc<Alert>) -> bool {
```
Comment (doitian) [inline util/network-alert/src/notifier.rs]: Syntax error

```suggestion
    fn is_version_effective(&self, alert: &Arc<Alert>) -> bool {
```

File: util/network-alert/src/notifier.rs [hunk 3]
```diff
@@ -44,23 +67,10 @@ impl Notifier {
...
 
         // check conditions, figure out do we need to notice this alert
-        if alert
-            .min_version
-            .as_ref()
-            .map(|min_v| Version::parse(&self.client_version) < Version::parse(min_v))
-            == Some(true)
-        {
+        if !self.is_accept_version(&alert) {
             return;
```
Comment (doitian) [inline util/network-alert/src/notifier.rs]: Add a log here.

**Pull request merged**